[AEROGEAR-2374] Remove labeling task and add label to the secret instead

JameelB opened the pull request:
## Description
Add the serviceInstanceID label in the secret created during provision and remove the labeling task from provision.

## Progress
- [x] Remove the labeling task from the provision process
- [x] Add the label serviceInstanceID to the secret created during provision

## Verification Steps
- [x] Provision `Aerogear UPS` from the service catalog.
- [x] Verify that no labeling tasks are ran during the provision process.
- [x] Verify that the service is provisioned successfully.
- [x] Verify that the service instance has no `serviceName` label.
- [x] Verify that a `unified-push-server` secret has been created.
- [x] Verify that the label `serviceInstanceID` is included in this secrets yaml file.
- [x] Verify that the value of the `serviceInstanceID` matches with the service instance's `spec.externalID` value.

## Additional Notes
Related JIRA Ticket - https://issues.jboss.org/browse/AEROGEAR-2374

JameelB committed:
Remove labelling task and add label to the secret instead

Add the serviceInstanceID label in the secret created during provision
and remove the labelling task from provision.

dimitraz reviewed (approved):
verified :+1:

matzew reviewed (changes requested):
@JameelB can you please rebase this PR?

thanks!

JameelB commented:
@matzew rebased :)

psturc commented:
Test won't pass because there's no service instance present when you run `apb test`, which means that `_apb_service_instance_id` variable is not defined: https://jenkins-wendy.ci.feedhenry.org/blue/organizations/jenkins/AeroGearCatalog%2Funifiedpush-apb/detail/PR-30/1/pipeline#step-51-log-75

@pb82 any idea how we could solve/workaround this?

pb82 commented on roles/provision-unifiedpush-apb/templates/secret.yml.j2:
```diff
@@ -7,6 +7,7 @@ metadata:
     name: ups
     mobile: enabled
     serviceName: ups
+    serviceInstanceID: {{ _apb_service_instance_id }}
```
@JameelB as @psturc has mentioned, `_apb_service_instance_id` is not available when testing. You could create another template named secret_testing.yml.j2 that does not have the `serviceInstanceID` and then in your ansible role use something like this:

```
- name: "Create ups secret yaml file"
  template:
    src: secret.yml.j2
    dest: /tmp/secret.yaml
  when: _apb_service_instance_id is defined

- name: "Create ups testing secret yaml file"
  template:
    src: secret_testing.yml.j2
    dest: /tmp/secret.yaml
  when: _apb_service_instance_id is not defined
```

JameelB replied:
@pb82 Sure , going to add these changes in :+1: I believe @psturc is also looking into using the mobile-cli during the tests which may also eliminate this problem.

JameelB replied:
ping @maleck13 @dimitraz

JameelB committed:
Add secret template for testing that doesnt have the service instance id

The service instance is not available during testing which causes the
test to fail when creating the secret with the service instance id. A
secret template is created to be used for testing when the service
instance id is not defined.

pb82 commented on roles/provision-unifiedpush-apb/tasks/provision-ups.yml:
```diff
@@ -138,9 +138,21 @@
+- name: "Create ups secret-testing yaml file"
+  template:
+    src: secret-testing.yml.j2
+    dest: /tmp/secret-testing.yaml
```
I think you can use the same dest `/tmp/secret.yaml` in both cases. Then you could safe the extra `oc create` step that deals with the testing secret.
